src/core/public/http/external_url_service.test.ts

```diff
@@ -0,0 +1,291 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { ExternalUrlConfig } from 'src/core/server/types';
+
+import { injectedMetadataServiceMock } from '../mocks';
+
+import { ExternalUrlService } from './external_url_service';
+
+const setupService = ({
+  location,
+  serverBasePath,
+  policy,
+}: {
+  location: URL;
+  serverBasePath: string;
+  policy: ExternalUrlConfig['policy'];
+}) => {
+  const injectedMetadata = injectedMetadataServiceMock.createSetupContract();
+  injectedMetadata.getExternalUrlConfig.mockReturnValue({ policy });
+  injectedMetadata.getServerBasePath.mockReturnValue(serverBasePath);
+
+  const service = new ExternalUrlService();
+  return {
+    setup: service.setup({
+      injectedMetadata,
+      location,
+    }),
+  };
+};
+
+describe('External Url Service', () => {
+  describe('#validateUrl', () => {
```

Comment:
We should add some test cases with an allow-all policy (`{ allow: true }`).

Reply:
yeah the current validation prohibits this type of config, but I think it makes sense to allow this